homeassistant/components/volvooncall/__init__.py

```diff
@@ -131,46 +131,29 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     hass.data[DATA_KEY] = {}
 
-    coordinator = VolvoUpdateCoordinator(hass)
+    volvo_data = VolvoData(hass, connection, config)
 
-    data = hass.data[DATA_KEY] = VolvoData(hass, connection, config, coordinator)
+    hass.data[DATA_KEY] = VolvoUpdateCoordinator(hass, volvo_data)
 
-    _LOGGER.info("Logging in to service")
-    return await data.update()
-
-
-class VolvoUpdateCoordinator(DataUpdateCoordinator):
-    """Volvo coordinator."""
-
-    def __init__(self, hass):
-        """Initialize the data update coordinator."""
-
-        super().__init__(
-            hass,
-            _LOGGER,
-            name="volvooncall",
-            update_interval=DEFAULT_UPDATE_INTERVAL,
-        )
-
-    async def _async_update_data(self):
-        """Fetch data from API endpoint."""
-
-        async with async_timeout.timeout(10):
-            if not await self.hass.data[DATA_KEY].update():
-                raise UpdateFailed("Error communicating with API")
+    _LOGGER.debug("Logging in to service")
+    return await volvo_data.update()
 
 
 class VolvoData:
     """Hold component state."""
 
-    def __init__(self, hass, connection: Connection, config, coordinator):
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        connection: Connection,
+        config: ConfigType,
+    ) -> None:
         """Initialize the component state."""
         self.hass = hass
         self.vehicles = set()  # type: ignore[var-annotated]
         self.instruments = set()  # type: ignore[var-annotated]
```

Comment:
Here's the Instrument class:
https://github.com/molobrakos/volvooncall/blob/1ea4b5d595a1292fc9d2f635a8f34b486f9ee509/volvooncall/dashboard.py#L13